### PR TITLE
[CAZ-2860 | CAZ-2861] Detect possible fraud flow

### PR DESCRIPTION
**Pull request by tkaczu**
## Motivation and Context
https://eaflood.atlassian.net/browse/CAZ-2860
https://eaflood.atlassian.net/browse/CAZ-2861

## Description
Add flow for possible fraud detection.

## How Has This Been Tested?
Manually, RSpec added and updated.

## Screenshots (if appropriate):
Screenshot_20200618_083613
Screenshot_20200618_083552

## Checklist:
- [ ] It contains only changes required by issue (does not contain other PR)
- [ ] Includes the link to an issue (if apply)
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: app/controllers/vehicles_controller.rb [hunk 1]
```diff
@@ -96,6 +94,45 @@ def confirm_details
     redirect_to process_detail_form(form)
   end
 
+  ##
+  # Renders vehicle UK registered page
+  #
+  # ==== Path
+  #
+  #    GET /vehicles/uk_registerd
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb:102]: `/uk_registered_details`

File: app/controllers/vehicles_controller.rb [hunk 2]
```diff
@@ -96,6 +94,45 @@ def confirm_details
...
+  # * +vrn+ - vehicle registration number, required in the session
+  #
+  def uk_registered_details
+    process_details_action
+  end
+
+  ##
+  # Verifies if user confirms the vehicle's details.
+  # If yes, renders to {incorrect details}[rdoc-ref:VehiclesController.local_authority]
+  # If no, redirects to {incorrect details}[rdoc-ref:VehiclesController.incorrect_details]
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb:114]: `rdoc-ref:local_authority`
`rdoc-ref:incorrect_details`

File: app/controllers/vehicles_controller.rb [hunk 3]
```diff
@@ -96,6 +94,45 @@ def confirm_details
...
+  #
+  # ==== Path
+  #    POST /vehicles/confirm_details
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb:117]: `confirm_uk_registered_details`

File: app/controllers/vehicles_controller.rb [hunk 4]
```diff
@@ -96,6 +94,45 @@ def confirm_details
...
+  # * +confirm-vehicle+ - user confirmation of vehicle details, 'yes' or 'no', required in the query
+  #
+  # ==== Validations
+  # * +vrn+ - lack of VRN redirects to {enter_details}[rdoc-ref:VehiclesController.enter_details]
+  # * +confirm-vehicle+ - lack of it redirects to {incorrect details}[rdoc-ref:VehiclesController.incorrect_details]
+  #
+  def confirm_uk_registered_details
+    form = ConfirmationForm.new(confirmation)
+    unless form.valid?
+      return redirect_to uk_registered_details_vehicles_path, alert: form.errors.messages[:confirmation].first
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb:130]: can you move `form.errors.messages[:confirmation].first` to form method `error_message` ? and just `alert: form.error_message`

File: app/controllers/vehicles_controller.rb [hunk 5]
```diff
@@ -96,6 +94,45 @@ def confirm_details
...
+    end
+
+    redirect_to process_detail_form(form)
+  end
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb:134]:
```
if form.valid?
....
else
...
end
```

File: app/controllers/vehicles_controller.rb [hunk 6]
```diff
@@ -235,10 +279,13 @@ def country
     params['registration-country']
   end
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb:281]: rdoc

File: app/views/vehicles/incorrect_details.html.haml [hunk 7]
```diff
@@ -1,9 +1,12 @@
...
+- title_and_header = 'Incorrect vehicle details'
+- page_title(title_and_header)
+
+= link_to 'Back', @return_path, class: 'govuk-back-link', id: 'back-link'
+
 %main.govuk-main-wrapper#main-content{role: 'main'}
   .govuk-grid-row
     .govuk-grid-column-two-thirds
-      %h1.govuk-heading-l Incorrect vehicle details
+      %h1.govuk-heading-l= title_and_header
```
Comment (likeuwill) [inline app/views/vehicles/incorrect_details.html.haml:9]: goooood

File: app/views/vehicles/uk_registered_details.html.haml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+          %tr.govuk-table__row
+            %td.govuk-table__cell{scope: 'row'} Colour
+            %td.govuk-table__header{id: 'colour'}= @vehicle_details.colour
+          %tr.govuk-table__row
+            %td.govuk-table__cell{scope: 'row'} Fuel type
+            %td.govuk-table__header{id: 'fuel-type'}= @vehicle_details.fuel_type
+          - if @vehicle_details.taxi_private_hire_vehicle.eql?('Yes')
+            %tr.govuk-table__row
+              %td.govuk-table__cell{scope: 'row'} Taxi
+              %td.govuk-table__header{id: 'taxi'}= @vehicle_details.taxi_private_hire_vehicle
```
Comment (likeuwill) [inline app/views/vehicles/uk_registered_details.html.haml:57]: are you sure what we're checking what non uk vehicle can be a taxi?

File: spec/requests/vehicles_controller/confirm_uk_registered_details_spec.rb [hunk 9]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'VehicleCheckersController - POST #confirm_uk_registered_details', type: :request do
+  subject do
+    post confirm_uk_registered_details_vehicles_path, params: { 'confirm-vehicle' => confirmation }
```
Comment (likeuwill) [inline spec/requests/vehicles_controller/confirm_uk_registered_details_spec.rb:7]:
```suggestion
    post confirm_uk_registered_details_vehicles_path, params: { 'confirm-vehicle': confirmation }
```

File: app/controllers/vehicles_controller.rb (hunk 1 above)
Comment (tkaczu) [reply app/controllers/vehicles_controller.rb:102]: :+1:

File: app/controllers/vehicles_controller.rb (hunk 3 above)
Comment (tkaczu) [reply app/controllers/vehicles_controller.rb:117]: :+1:

File: app/controllers/vehicles_controller.rb (hunk 2 above)
Comment (tkaczu) [reply app/controllers/vehicles_controller.rb:114]: :+1:

File: app/controllers/vehicles_controller.rb (hunk 4 above)
Comment (tkaczu) [reply app/controllers/vehicles_controller.rb:130]: :+1:

File: app/controllers/vehicles_controller.rb (hunk 5 above)
Comment (tkaczu) [reply app/controllers/vehicles_controller.rb:134]: :+1:

File: app/controllers/vehicles_controller.rb (hunk 6 above)
Comment (tkaczu) [reply app/controllers/vehicles_controller.rb:281]: :+1:

File: app/views/vehicles/uk_registered_details.html.haml (hunk 8 above)
Comment (tkaczu) [reply app/views/vehicles/uk_registered_details.html.haml:57]: As we talk after daily today - this should stay ;)

File: spec/requests/vehicles_controller/confirm_uk_registered_details_spec.rb (hunk 9 above)
Comment (tkaczu) [reply spec/requests/vehicles_controller/confirm_uk_registered_details_spec.rb:7]: :+1:

File: app/controllers/vehicles_controller.rb [hunk 10]
```diff
@@ -87,19 +87,19 @@ def details
   #
   def confirm_details
     form = ConfirmationForm.new(confirmation)
-    unless form.valid?
-      return redirect_to details_vehicles_path, alert: form.errors.messages[:confirmation].first
+    if form.valid?
+      redirect_to process_detail_form(form)
+    else
+      redirect_to details_vehicles_path, alert: form.error_message
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb:93]: please use `render`, e.g.
```
      flash.now[:alert] = I18n.t('select_weekly_period')
      render :select_weekly_period
```

File: app/controllers/vehicles_controller.rb [hunk 11]
```diff
@@ -126,11 +126,11 @@ def uk_registered_details
   #
   def confirm_uk_registered_details
     form = ConfirmationForm.new(confirmation)
-    unless form.valid?
-      return redirect_to uk_registered_details_vehicles_path, alert: form.errors.messages[:confirmation].first
+    if form.valid?
+      redirect_to process_detail_form(form)
+    else
+      redirect_to uk_registered_details_vehicles_path, alert: form.error_message
```
Comment (likeuwill) [inline app/controllers/vehicles_controller.rb:132]: same, use `render`

**Pull request merged**